server/container/guice/mailet/src/test/java/org/apache/james/utils/GuilceMailetLoaderTest.java

```diff
@@ -124,38 +71,41 @@ public void getMailetShouldThrowOnBadType() throws Exception {
 
     @Test
     public void getMailetShouldLoadClassWhenInIncludedJars() throws Exception {
-        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(Guice.createInjector(), CLASSPATH_FILE_SYSTEM);
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(injector,
+            new ExtendedClassLoader(CLASSPATH_FILE_SYSTEM));
 
         Mailet mailet = guiceMailetLoader.getMailet(FakeMailetConfig.builder()
-            .mailetName("org.apache.transport.mailets.AIMailet")
+            .mailetName("CustomMailet")
             .mailetContext(FakeMailContext.defaultContext())
             .build());
 
         assertThat(mailet.getClass().getCanonicalName())
-            .isEqualTo("org.apache.transport.mailets.AIMailet");
+            .isEqualTo("org.apache.james.transport.mailets.CustomMailet");
     }
 
     @Test
     public void getMailedShouldShouldRecursivelyIncludeJar() throws Exception {
-        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(Guice.createInjector(), RECURSIVE_CLASSPATH_FILE_SYSTEM);
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(injector,
+            new ExtendedClassLoader(RECURSIVE_CLASSPATH_FILE_SYSTEM));
 
         Mailet mailet = guiceMailetLoader.getMailet(FakeMailetConfig.builder()
-            .mailetName("org.apache.transport.mailets.AIMailet")
+            .mailetName("CustomMailet")
             .mailetContext(FakeMailContext.defaultContext())
             .build());
 
         assertThat(mailet.getClass().getCanonicalName())
-            .isEqualTo("org.apache.transport.mailets.AIMailet");
+            .isEqualTo("org.apache.james.transport.mailets.CustomMailet");
     }
 
     @Test
     public void getMailetShouldThrowOnUnknownMailet() throws Exception {
-        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(Guice.createInjector(), CLASSPATH_FILE_SYSTEM);
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(injector,
+            new ExtendedClassLoader(CLASSPATH_FILE_SYSTEM));
 
         expectedException.expect(MessagingException.class);
 
         guiceMailetLoader.getMailet(FakeMailetConfig.builder()
-            .mailetName("org.apache.transport.mailets.Unknown")
+            .mailetName("org.apache.james.transport.mailets.Unknown")
             .mailetContext(FakeMailContext.defaultContext())
             .build());
     }
```

Comment:
Can you please also test with a mailet in an other package than org.apache.james?